### PR TITLE
Refactor namings around engine and registers

### PR DESCRIPTION
**Pull request by mathetake**
- Renamed `*BaseStack*` -> `*StackBase*`
- Prefixed all the reserved registers with `reservedRegisterFor`

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: wasm/jit/jit_amd64.go [hunk 1]
```diff
@@ -1487,7 +1475,7 @@ func (b *amd64Builder) assignRegisterToValue(loc *valueLocation, reg int16) {
 	prog := b.newProg()
 	prog.As = x86.AMOVQ
 	prog.From.Type = obj.TYPE_MEM
-	prog.From.Reg = cachedStackBasePointerReg
+	prog.From.Reg = reservedRegisterForStackBasePointer
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64.go:1478]: ⚡

File: wasm/jit/jit_value_location_amd64.go [hunk 2]
```diff
@@ -214,9 +224,9 @@ func (s *valueLocationStack) takeFreeRegister(tp generalPurposeRegisterType) (re
 	var targetRegs []int16
 	switch tp {
 	case gpTypeFloat:
-		targetRegs = gpFloatRegisters
+		targetRegs = generalPurposeFloatRegisters
 	case gpTypeInt:
-		targetRegs = gpIntRegisters
+		targetRegs = unreservedGeneralPurposeIntRegisters
```
Comment (codefromthecrypt) [inline wasm/jit/jit_value_location_amd64.go:229]: so that's what gp is? I thought it was "general practitioner" 👨‍⚕️

**Review by codefromthecrypt — APPROVED** [review]
thanks a lot

File: wasm/jit/jit_value_location_amd64.go (hunk 2 above)
Comment (mathetake) [reply wasm/jit/jit_value_location_amd64.go:229]: removed all the usage of "gp": https://github.com/tetratelabs/wazero/pull/84/commits/43f965864a047b0c81f9e3c1e6475b24dc79f553 👯

**Pull request merged**